Servers/UI/OJS.Servers.Ui/ClientApp/src/components/search/SearchSection.tsx

```diff
@@ -0,0 +1,172 @@
+        setItemsSearchError(null);
+        getItemSearchResults();
+        setIsLoading(false);
+    }, [ getSearchResultsParams, getItemSearchResults ]);
+
+    useEffect(() => {
+        if (isNil(itemSearchResultsData) || isEmpty(itemSearchResultsData)) {
+            return;
+        }
+
+        if (!isNil(getItemSearchResultsError)) {
+            setItemsSearchError(getItemSearchResultsError);
+            setSearchingError(getItemSearchResultsError);
+
+            return;
+        }
+
+        const searchResult = itemSearchResultsData as IPagedResultType<IGenericSearchResponseModel<T>>;
+
+        const { items: searchResponseData } = searchResult;
+        if (isNil(searchResponseData) || isEmpty(searchResponseData)) {
+            return;
+        }
+
+        const foundItems = Object.values(first(searchResponseData) as IGenericSearchResponseModel<T>);
+
+        setSearchedItems(foundItems[0] as T[]);
+        setItemsSearchError(null);
+        setSearchingError(null);
+        setSearchResults(searchResult);
+    }, [ getItemSearchResultsError, itemSearchResultsData, searchCategory, searchResults,
+        searchedItems, setSearchingError ]);
+
+    const handlePageChange = useCallback((newPage : number) => {
+        setCurrentItemsPage(newPage);
+    }, []);
+
+    const renderNoResultsFound = useCallback(
+        () => (
+            <div className={styles.headingSearch}>
+                <Heading
+                  type={HeadingType.secondary}
+                  className={styles.heading}
+                >
+                    {searchCategory}
+                    :
+                </Heading>
+                <Heading
+                  type={HeadingType.secondary}
+                  className={styles.searchHeading}
+                >
+                    No results found for
+                    {' '}
+                    {searchCategory.toLowerCase()}
+                </Heading>
+            </div>
+        ),
+        [ searchCategory ],
+    );
+
+    const renderItems = useCallback(
+        (items: T[]) => (
+            <>
+                <Heading
+                  type={HeadingType.secondary}
+                  className={styles.heading}
+                >
+                    {searchCategory}
+                    :
+                </Heading>
+                <List
+                  values={items}
+                  itemFunc={renderItem}
+                  className={styles.items}
+                  itemClassName={styles.contestItem}
+                  orientation={Orientation.horizontal}
+                  wrap
+                />
+                <PaginationControls
+                  count={searchResults.pagesCount}
+                  page={currentItemsPage}
+                  onChange={handlePageChange}
+                />
+            </>
+        ),
+        [ searchResults, currentItemsPage, handlePageChange, renderItem, searchCategory ],
+    );
+
+    return (
+        <div>
```

Comment:
Isnt this resulting in an empty `div` if nothing is returned in `{ }`?

Reply:
Refactored - removed the div tag. Technically it will never reach this check if there is error since SearchPage will not render SearchSections and render error message instead.